src/plugins/orangehrmCorporateDirectoryPlugin/Api/EmployeeDirectoryAPI.php

```diff
@@ -0,0 +1,243 @@
+    public const FILTER_JOB_TITLE_ID = 'jobTitleId';
+    public const FILTER_LOCATION_ID = 'locationId';
+    public const FILTER_MODEL = 'model';
+
+    public const MODEL_DEFAULT = 'default';
+    public const MODEL_DETAILED = 'detailed';
+    public const MODEL_MAP = [
+        self::MODEL_DEFAULT => EmployeeDirectoryModel::class,
+        self::MODEL_DETAILED => EmployeeDirectoryDetailedModel::class,
+    ];
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResourceResult
+    {
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+        $employee = $this->getEmployeeDirectoryService()->getEmployeeDirectoryDao()->getEmployeeByEmpNumber($empNumber);
+        $this->throwRecordNotFoundExceptionIfNotExist($employee, Employee::class);
+
+        return new EndpointResourceResult($this->getModelClass(), $employee);
+    }
+
+    /**
+     * @return EmployeeDirectoryService
+     */
+    public function getEmployeeDirectoryService(): EmployeeDirectoryService
+    {
+        return new EmployeeDirectoryService();
+    }
+
+    /**
+     * @return string
+     */
+    protected function getModelClass(): string
+    {
+        $model = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_MODEL,
+            self::MODEL_DEFAULT
+        );
+        return self::MODEL_MAP[$model];
+    }
+
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_EMP_NUMBER,
+                new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)
+            ),
+            $this->getModelParamRule(),
+        );
+    }
+
+    /**
+     * @return ParamRule
+     */
+    protected function getModelParamRule(): ParamRule
+    {
+        return $this->getValidationDecorator()->notRequiredParamRule(
+            new ParamRule(
+                self::FILTER_MODEL,
+                new Rule(Rules::IN, [array_keys(self::MODEL_MAP)])
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointCollectionResult
+    {
+        $employeeDirectoryParamHolder = new EmployeeDirectorySearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeDirectoryParamHolder);
+
+        $empNumber = $this->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_EMP_NUMBER
+        );
+        if (!is_null($empNumber)) {
+            $employeeDirectoryParamHolder->setEmpNumbers([$empNumber]);
+        } else {
+            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
+            $employeeDirectoryParamHolder->setEmpNumbers($accessibleEmpNumbers);
+        }
```

Comment:
```suggestion
        }
```
may have to remove else condition since any user role can access any employee here

Reply:
done